frontend/zulu/src/components/Map/MapGeojsonMarkers.jsx

```diff
@@ -61,18 +71,76 @@ class MapComponent extends React.Component {
           );
 
     }
+
+    handleChange(event) {
+      var target = event.target;
+      var value = target.value;
+      var name = target.name;
+
+      this.setState({
+        [name]: value
+      });
+    }
+
+    async handleSubmit(event) {
+      event.preventDefault();
+      let file = this.fileInput?.current?.files[0];
+      let hasFile = (typeof file !== 'undefined');
+
+      let imageId = null;
+
+      if (hasFile) { // todo get photo id
+        let formData = new FormData();
+        formData.append('image', file);
+
+        await StoryService.postImage(
+          formData
+        ).then(function (response) {
+            //handle success
+            imageId = response.data.image_id;
+            console.log(response);
+        })
+        .catch(function (response) {
+            //handle error
+            console.log(response);
+            alert('there was a problem adding your image, please try again soon.');
+        });
+      }
+
+      let lng = this.state.lng;
+      let lat = this.state.lat;
+      let title = this.state.storyTitle;
+      let content = this.state.storyBody;
+
+      StoryService.postUserStory(
+        lng, lat, title, content, imageId
```

Comment:
I wonder if this would somehow propogate to the backend and add 'image_id's with nulls when not uploading an image -  ot sure if all code is "safe" here - worth checking if you didn't yet.